app/src/main/java/org/mozilla/tv/firefox/search/SearchEngineProviderWrapper.kt

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.search
+
+import android.content.Context
+import android.util.Log
+import java.util.Locale
+import mozilla.components.browser.search.SearchEngine
+import mozilla.components.browser.search.provider.AssetsSearchEngineProvider
+import mozilla.components.browser.search.provider.SearchEngineProvider
+import mozilla.components.browser.search.provider.localization.SearchLocalizationProvider
+
+private const val LOGTAG = "Search"
+
+/**
+ * Wraps an [AssetsSearchEngineProvider] to allow for replacements of
+ * search plugins.
+ *
+ * @property replacements a map specifying which plugins to replace e.g.
+ * mapOf("google" to "google-fftv") to replace google with google-fftv.
+ */
+class SearchEngineProviderWrapper(private val replacements: Map<String, String>) : SearchEngineProvider {
+
+    val myLocalizationProvider = object : SearchLocalizationProvider() {
+        override val language: String
+            get() = Locale.getDefault().language
+
+        override val country: String
+            get() = Locale.getDefault().country
+
+        override val region: String? = country
+    }
+
+    private val inner = AssetsSearchEngineProvider(
+        localizationProvider = myLocalizationProvider,
+        additionalIdentifiers = replacements.values.toList()
+    )
+
+    override suspend fun loadSearchEngines(context: Context): List<SearchEngine> {
```

Comment:
Optional: this seems like a method that would benefit from having a few tests.

Reply:
I'm not sure how to do that, would need some help.

Reply:
Opened a PR into your branch that adds a few basic unit tests.  If you think of more that would be appropriate, feel free to add them in!  If not, don't worry about it, it's not a super complex use case.

https://github.com/mkaply/firefox-tv/pull/1